benchmarks/kubectl-mtb/test/benchmarks/block_other_tenant_resources/block_other_tenant_resources.go

```diff
@@ -0,0 +1,94 @@
+package blockothertenantresources
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/test/utils"
+	"sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/types"
+
+	"sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/bundle/box"
+	"sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/pkg/benchmark"
+	"sigs.k8s.io/multi-tenancy/benchmarks/kubectl-mtb/test"
+)
+
+var verbs = []string{"get", "update"}
+
+var b = &benchmark.Benchmark{
+
+	PreRun: func(options types.RunOptions) error {
+
+		return nil
+	},
+
+	Run: func(options types.RunOptions) error {
+		var primaryNamespaceResources []utils.GroupResource
+
+		lists, err := options.KClient.Discovery().ServerPreferredResources()
+		if err != nil {
+			options.Logger.Debug(err.Error())
+			return err
+		}
+
+		for _, list := range lists {
+			if len(list.APIResources) == 0 {
+				continue
+			}
+			gv, err := schema.ParseGroupVersion(list.GroupVersion)
+			if err != nil {
+				continue
+			}
+			for _, resource := range list.APIResources {
+				if len(resource.Verbs) == 0 {
+					continue
+				}
+
+				if resource.Namespaced {
+					continue
+				}
+				primaryNamespaceResources = append(primaryNamespaceResources, utils.GroupResource{
+					APIGroup:    gv.Group,
+					APIResource: resource,
+				})
+			}
+		}
+
+		for _, resource := range primaryNamespaceResources {
+			for _, verb := range verbs {
+				access, msg, err := utils.RunAccessCheck(options.OClient, options.TenantNamespace, resource, verb)
+				if err != nil {
+					options.Logger.Debug(err.Error())
+					return err
+				}
+				if access {
+					return fmt.Errorf(msg)
+				}
+			}
+		}
+
+		for _, resource := range primaryNamespaceResources {
+			for _, verb := range verbs {
+				access, msg, err := utils.RunAccessCheck(options.TClient, options.OtherNamespace, resource, verb)
+				if err != nil {
+					options.Logger.Debug(err.Error())
+					return err
+				}
+				if access {
+					return fmt.Errorf(msg)
+				}
+			}
+		}
```

Comment:
I think these two for loops can be clubbed into a single for loop.

Reply:
Yeah this can be made a separate function, right now this particular piece of duplicated at three places

Reply:
Please refactor as suggested.